File: pbmx_ffi/java/io/rmf/pbmx/ffi/RawFingerprint.java
package io.rmf.pbmx.ffi;

import com.sun.jna.Structure;
import java.util.List;
import java.util.Arrays;

public class RawFingerprint extends Structure {
    public static class ByValue extends RawFingerprint implements Structure.ByValue {}

    public byte[] bytes = new byte[20];

    public ByValue val() {
        ByValue value = new ByValue();
        value.bytes = this.bytes;
        return value;
    }

    public RawFingerprint ref() {
        RawFingerprint s = new RawFingerprint();
        s.bytes = this.bytes;
        return s;
    }

    @Override
    protected List getFieldOrder() {
        return Arrays.asList(new String[] { "bytes" });
    }
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/ffi/RawFingerprint.kt
package io.rmf.pbmx.ffi

import com.sun.jna.Structure

open class RawFingerprint : Structure() {

    @JvmField var bytes = ByteArray(20)

    class ByValue : RawFingerprint(), Structure.ByValue

    fun value(): ByValue {
        val value = ByValue()
        value.bytes = this.bytes
        return value
    }

    override fun getFieldOrder(): List<String> {
        return listOf("bytes")
    }
}
